models/lpjguess/R/write_restart.LPJGUESS.R

```diff
@@ -0,0 +1,115 @@
+##' write_restart.LPJGUESS
+##'
+##' Write restart files for LPJGUESS
+##' new.state includes X (AGB.pft) from Analysis
+##' new.params includes LPJGUESS_state
+##' 
+##' @param outdir output directory
+##' @param runid run ID
+##' @param start.time start date and time for each SDA ensemble
+##' @param stop.time stop date and time for each SDA ensemble
+##' @param settings PEcAn settings object
+##' @param new.state analysis state vector
+##' @param RENAME flag to either rename output file or not
+##' @param new.params list of parameters to convert between different states 
+##' @param inputs list of model inputs to use in write.configs.SIPNET
+##' @param verbose decide if we want to print the runid
+##' 
+##' @return NONE
+##'
+##' @importFrom dplyr %>%
+##' @export
+##' @author Yinghao Sun
+write_restart.LPJGUESS <- function(outdir, runid,
+                                   start.time, stop.time, settings,
+                                   new.state, RENAME = TRUE,
+                                   new.params, inputs = NULL, verbose = FALSE){
+  
+  rundir <- settings$host$rundir
+  variables <- colnames(new.state)
+  
+  ## ---- Rename old output, remove old clim ----
+  if (RENAME) {
+    file.rename(file.path(outdir, runid, "lpjguess.out"),
+                file.path(outdir, runid, paste0("lpjguess.", as.Date(start.time), ".out")))
+    system(paste("rm", file.path(rundir, runid, "lpjguess.clim")))
+  } else {
+    print(paste("Files not renamed -- Need to rerun timestep", start.time, "before next time step"))
+  }
+  
+  settings$run$start.date <- start.time
+  settings$run$end.date <- stop.time
+  
+  ## ---- Pull old state ----
+  if (is.null(new.params$LPJGUESS_state))
+    PEcAn.logger::logger.severe("LPJGUESS_state missing in new.params")
+  # new.params$LPJGUESS_state include state, pos_list, siz_list
+  Gridcell <- new.params$LPJGUESS_state$state
+  pos_list <- new.params$LPJGUESS_state$pos_list
+  siz_list <- new.params$LPJGUESS_state$siz_list
+  
+
+  
+  ## ---- Build PFT parameter table from new.params ----
```

Comment:
Are these parameters that are needed to be able to write the restart file (e.g. convert new.state to model specific outputs) or just to parameters needed to write the param file? Presumably the latter has not changed from what was written during write.configs. For the former, the custom has been to pass these through new.params. It makes me nervous to see a whole bunch of parameters hard-coded in write_restart